### PR TITLE
chore: fix double notification routing and segment callback

### PR DESCRIPTION
**Pull request by saeedbashir**
This PR fixes the issue with push notifications.

Issue details: https://github.com/openedx/openedx-app-ios/issues/472

### CODE REVIEW

File: OpenEdX/Managers/PushNotificationsManager/PushNotificationsManager.swift [hunk 1]
```diff
@@ -29,18 +29,28 @@ class PushNotificationsManager: NSObject {
     private let deepLinkManager: DeepLinkManager
     private let storage: CoreStorage
     private let api: API
+    private let segemntService: SegmentAnalyticsService?
```
Comment (volodymyr-chekyrta) [inline OpenEdX/Managers/PushNotificationsManager/PushNotificationsManager.swift:32]: typo segemntService -> segmentService

File: OpenEdX/Managers/PushNotificationsManager/PushNotificationsManager.swift [hunk 2]
```diff
@@ -29,18 +29,28 @@ class PushNotificationsManager: NSObject {
...
     }
     
     // Init manager
-    public init(deepLinkManager: DeepLinkManager, storage: CoreStorage, api: API, config: ConfigProtocol) {
+    public init(deepLinkManager: DeepLinkManager,
+                storage: CoreStorage,
+                api: API,
+                config: ConfigProtocol,
+                segmentService: SegmentAnalyticsService?
+    ) {
```
Comment (volodymyr-chekyrta) [inline OpenEdX/Managers/PushNotificationsManager/PushNotificationsManager.swift:48]:
```suggestion
    public init(
        deepLinkManager: DeepLinkManager,
        storage: CoreStorage,
        api: API,
        config: ConfigProtocol,
        segmentService: SegmentAnalyticsService?
    ) {
```

File: OpenEdX/DI/AppAssembly.swift [hunk 3]
```diff
@@ -177,7 +177,8 @@ class AppAssembly: Assembly {
                 deepLinkManager: r.resolve(DeepLinkManager.self)!,
                 storage: r.resolve(CoreStorage.self)!,
                 api: r.resolve(API.self)!,
-                config: r.resolve(ConfigProtocol.self)!
+                config: r.resolve(ConfigProtocol.self)!,
+                segmentService: r.resolve(SegmentAnalyticsService.self)
```
Comment (volodymyr-chekyrta) [inline OpenEdX/DI/AppAssembly.swift:181]: Could we use an Analytics abstraction and not rely on the Segment dependency here?
We are going to migrate all third-party services to plugins, and this dependency could make it harder.

**Review by rnr — APPROVED** [review]

**Review by volodymyr-chekyrta — APPROVED** [review]
LGTM

**Pull request merged**